Muehle_Algo.ipynb

```diff
@@ -0,0 +1,353 @@
+{
+ "metadata": {
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.7.10-final"
+  },
+  "orig_nbformat": 2,
+  "kernelspec": {
+   "name": "python3",
+   "display_name": "Python 3.7.10 64-bit ('ai': conda)",
+   "metadata": {
+    "interpreter": {
+     "hash": "6aef3afc8e1df4532fc05871b69d433488f1644eaca8e0d1af5a2e1125595c9e"
+    }
+   }
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 2,
+ "cells": [
+  {
+   "source": [
+    "# Algorithmen\n",
+    "\n"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%run ./Muehle_Logic.ipynb\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%run ./Muehle_Utilities.ipynb"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%run ./Muehle_Heuristik.ipynb"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "START_STATE = [[9, 9], [          # Anzahl zu setzender Steine (Spieler_1 (Weiß), Spieler_2 (Braun))\n",
+    "            [0, 0, 0, 0, 0, 0, 0, 0], # ring 0\n",
+    "            [0, 0, 0, 0, 0, 0, 0, 0], # ring 1\n",
+    "            [0, 0, 0, 0, 0, 0, 0, 0]  # ring 2\n",
+    "      ]]\n",
+    "board7 = [[0, 0],\n",
+    "         [[0, 0, 1, 2, 0, 1, 0, 0],\n",
+    "          [2, 0, 0, 0, 0, 0, 0, 0],\n",
+    "          [0, 2, 0, 0, 0, 2, 0, 0]]]\n",
+    "\n",
+    "def to_tupel(State):\n",
+    "    return ((State[0][0], State[0][1]), (\n",
+    "                (State[1][0][0],State[1][0][1],State[1][0][2],State[1][0][3],State[1][0][4],State[1][0][5],State[1][0][6],State[1][0][7]),\n",
+    "                (State[1][1][0],State[1][1][1],State[1][1][2],State[1][1][3],State[1][1][4],State[1][1][5],State[1][1][6],State[1][1][7]),\n",
+    "                (State[1][2][0],State[1][2][1],State[1][2][2],State[1][2][3],State[1][2][4],State[1][2][5],State[1][2][6],State[1][2][7])\n",
+    "            ))\n",
+    "\n",
+    "def to_list(State):\n",
+    "    return [[State[0][0], State[0][1]], [\n",
+    "                [State[1][0][0],State[1][0][1],State[1][0][2],State[1][0][3],State[1][0][4],State[1][0][5],State[1][0][6],State[1][0][7]],\n",
+    "                [State[1][1][0],State[1][1][1],State[1][1][2],State[1][1][3],State[1][1][4],State[1][1][5],State[1][1][6],State[1][1][7]],\n",
+    "                [State[1][2][0],State[1][2][1],State[1][2][2],State[1][2][3],State[1][2][4],State[1][2][5],State[1][2][6],State[1][2][7]]\n",
+    "    ]]\n",
```

Comment:
1. State -> state
2. Die Funktionen könnten wir eigentlich auch in das Utilities-Notebook packen, oder?